packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -114,19 +138,71 @@ class CupertinoCheckbox extends StatefulWidget {
   /// ```
   final ValueChanged<bool?>? onChanged;
 
+  /// The cursor for a mouse pointer when it enters or is hovering over the
+  /// widget.
+  ///
+  /// If [mouseCursor] is a [WidgetStateMouseCursor],
+  /// [WidgetStateMouseCursor.resolve] is used for the following [WidgetState]s:
+  ///
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// When [value] is null and [tristate] is true, [WidgetState.selected] is
+  /// included as a state.
+  ///
+  /// If null, then [SystemMouseCursors.basic] is used.
+  final MouseCursor? mouseCursor;
+
   /// The color to use when this checkbox is checked.
   ///
   /// Defaults to [CupertinoColors.activeBlue].
+  ///
+  /// If [fillColor] returns a non-null color in the [WidgetState.selected]
+  /// state, it will be used instead of this color.
   final Color? activeColor;
 
   /// The color used if the checkbox is inactive.
   ///
   /// By default, [CupertinoColors.inactiveGray] is used.
   final Color? inactiveColor;
 
+  /// The color that fills the checkbox, in all [WidgetState]s.
+  ///
+  /// Resolves in the following states:
+  ///  * [WidgetState.selected].
+  ///  * [WidgetState.hovered].
+  ///  * [WidgetState.focused].
+  ///  * [WidgetState.disabled].
+  ///
+  /// {@tool snippet}
+  /// This example resolves the [fillColor] based on the current [WidgetState]
+  /// of the [CupertinoCheckbox], providing a different [Color] when it is
+  /// [WidgetState.disabled].
+  ///
+  /// ```dart
+  /// CupertinoCheckbox(
+  ///   value: true,
+  ///   onChanged: (_){},
+  ///   fillColor: WidgetStateProperty.resolveWith<Color>((Set<WidgetState> states) {
+  ///     if (states.contains(WidgetState.disabled)) {
+  ///       return Colors.orange.withOpacity(.32);
+  ///     }
+  ///     return Colors.orange;
+  ///   })
+  /// )
+  /// ```
+  /// {@end-tool}
+  ///
+  /// If null, then the value of [activeColor] is used in the selected
+  /// state, and [CupertinoColors.white] is used in the unselected state.
+  final WidgetStateProperty<Color?>? fillColor;
```

Comment:
Why does this have to resolve to `Color?` instead of `Color`? Sorry if there was a discussion before. But I think we should just make it `WidgetStateProperty<Color>? fillColor;` and having a non-null `fillColor` invalidates all other legacy color properties. When will a user assign a `fillColor` that resolves to non-null except for `selected` and also assigns an `activeColor`?

Reply:
It exists for `.adaptive` compatibility. Material `Checkbox`'s `fillColor` is also a `WidgetStateProperty<Color?>?`.

Reply:
Looks like this was removed, marking as resovled.